src/app/shared/context.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { Context } from './../models/context';
+import { DummyService } from './../dummy/dummy.service';
+import { Broadcaster } from '../shared/broadcaster.service';
+
+
+/*
+ * A shared service that manages the users current context. The users context is defined as the
+ * entity (user or org) and space that they are operating on.
+ * 
+ */
+@Injectable()
+export class ContextService {
+
+  private _current: Context;
+  constructor(
+    private dummy: DummyService,
+    private router: Router,
+    private broadcaster: Broadcaster
+  ) {
+    // Listen for any context refreshes requested by the app
+    this.broadcaster.on<string>('refreshContext').subscribe(message => {
+      this.computeContext();
+    });
+    // Compute the initial context
+    this.computeContext();
+  }
+
+  get current(): Context {
+    return this._current;
+  }
+
+  private computeContext() {
+    // Find the most specific context menu path and display it
+    let c;
+    for (let m of this.dummy.contexts) {
+      if (this.router.url.startsWith(m.path)) {
+        if (c == null || m.path.length > c.path.length) {
```

Comment:
It does not appear that you initialize or set 'c' and therefor it would always be null.